src/main/serialport.ts

```diff
@@ -0,0 +1,348 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-4-Clause
+ */
+
+import type { AutoDetectTypes, SetOptions } from '@serialport/bindings-cpp';
+import { Renderer, WebContents } from 'electron';
+import { SERIALPORT_CHANNEL } from 'pc-nrfconnect-shared/main';
+import { SerialPort, SerialPortOpenOptions } from 'serialport';
+
+export type Renderer = Pick<WebContents, 'on' | 'send' | 'id'>;
+
+/**
+ *  serialPorts: {
+
+        '/dev/Robot': {
+            renderers: Renderer[],
+            serialPort: SerialPort,
+        },
+
+        '/dev/Rowboat': {
+            renderers: Renderer[],
+            serialPort: SerialPort,
+        }
+    }
+*/
+type OpenPort = {
+    serialPort: SerialPort;
+    renderers: Renderer[];
+    settingsLocked: boolean;
+    opening: boolean;
+};
+
+const serialPorts = new Map<string, OpenPort>();
```

Comment:
One concern is that on windows COM1 and com1 are both valid and the same port. but a different key for MAP.

Consider wrapping the set and get of the map to ensure that we don't have any case issues

Reply:
Fixed in last commit #79d41d03a